core/core.go

```diff
@@ -2,73 +2,88 @@ package core
 
 import (
 	"context"
-	"os"
-	"path"
+	"errors"
+	"net"
+	"net/http"
+	"strings"
 	"time"
 
-	"github.com/google/uuid"
+	"github.com/dgrijalva/jwt-go"
+	grpcm "github.com/grpc-ecosystem/go-grpc-middleware"
 	auth "github.com/grpc-ecosystem/go-grpc-middleware/auth"
-	"github.com/ipfs/go-datastore"
-	badger "github.com/ipfs/go-ds-badger"
+	"github.com/improbable-eng/grpc-web/go/grpcweb"
 	httpapi "github.com/ipfs/go-ipfs-http-client"
 	logging "github.com/ipfs/go-log"
+	"github.com/libp2p/go-libp2p-core/crypto"
 	"github.com/libp2p/go-libp2p-core/peer"
 	ma "github.com/multiformats/go-multiaddr"
-	fc "github.com/textileio/filecoin/api/client"
-	threadsapi "github.com/textileio/go-threads/api"
-	threadsclient "github.com/textileio/go-threads/api/client"
+	filc "github.com/textileio/filecoin/api/client"
+	dbapi "github.com/textileio/go-threads/api"
+	threads "github.com/textileio/go-threads/api/client"
+	dbpb "github.com/textileio/go-threads/api/pb"
 	"github.com/textileio/go-threads/broadcast"
-	serviceapi "github.com/textileio/go-threads/service/api"
-	s "github.com/textileio/go-threads/store"
-	"github.com/textileio/go-threads/util"
-	"github.com/textileio/textile/api"
+	tc "github.com/textileio/go-threads/common"
+	"github.com/textileio/go-threads/core/thread"
+	netapi "github.com/textileio/go-threads/net/api"
+	netpb "github.com/textileio/go-threads/net/api/pb"
+	tutil "github.com/textileio/go-threads/util"
+	"github.com/textileio/textile/api/buckets"
+	bpb "github.com/textileio/textile/api/buckets/pb"
+	"github.com/textileio/textile/api/common"
+	"github.com/textileio/textile/api/hub"
+	hpb "github.com/textileio/textile/api/hub/pb"
+	"github.com/textileio/textile/api/users"
+	upb "github.com/textileio/textile/api/users/pb"
 	c "github.com/textileio/textile/collections"
 	"github.com/textileio/textile/dns"
 	"github.com/textileio/textile/email"
 	"github.com/textileio/textile/gateway"
+	"github.com/textileio/textile/util"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
 var (
 	log = logging.Logger("core")
-)
 
-// clientReqKey provides a concrete type for client request context values.
-type clientReqKey string
+	ignoreMethods = []string{
+		"/hub.pb.API/Signup",
+		"/hub.pb.API/Signin",
+		"/hub.pb.API/IsUsernameAvailable",
```

Comment:
Method are wide open to the public.

Reply:
Any reason we might want to do something like rate limit these? To avoid blasting us with new signups?

Reply:
Yeah good call. I'll create an issue.